src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/prison/UserControllerIntTest.kt

```diff
@@ -533,6 +534,67 @@ class UserControllerIntTest : IntegrationTestBase() {
     }
   }
 
+  @Nested
+  inner class FindUserByUserName {
+    private val username = "NUSER_GEN"
+
+    @Test
+    fun `access forbidden when no authority`() {
+      webTestClient.get().uri("/prisonusers/$username")
+        .exchange()
+        .expectStatus().isUnauthorized
+    }
+
+    @Test
+    fun `access forbidden when no role`() {
+      webTestClient.get().uri("/prisonusers/$username")
+        .headers(setAuthorisation(roles = listOf()))
+        .exchange()
+        .expectStatus().isForbidden
+    }
+
+    @Test
+    fun `access forbidden when wrong role`() {
+      webTestClient.get().uri("/prisonusers/$username")
+        .headers(setAuthorisation(roles = listOf("ROLE_WRONG_ROLE")))
+        .exchange()
+        .expectStatus().isForbidden
+    }
+
+    @Test
+    fun `user not found`() {
+      nomisApiMockServer.stubGetFail("/users/$username", NOT_FOUND)
+      webTestClient.get().uri("/prisonusers/$username")
+        .headers(setAuthorisation(roles = listOf("ROLE_MAINTAIN_ACCESS_ROLES_ADMIN")))
+        .exchange()
+        .expectStatus().isNotFound
+
+      nomisApiMockServer.verify(
+        getRequestedFor(urlEqualTo("/users/$username")),
+      )
+    }
+
```

Comment:
Looks like we're still missing a test when the username contains "@" - to ensure it fails gracefully

Reply:
ok, will have a look , thanks